### DISCUSSION TITLE
why focus:ring can work success on windows browser,  but fail work on ios/iphone/safari ?

### DISCUSSION THREAD

**Discussion by e2366705** [tailwindlabs/tailwindcss#5998] [Help] [answered]
when i click the button, i want a ring besiege the button ; 
don't show the ring when i not click the button ,  

but it's only work on windows browser,  not work in ios/iphone/safari 

i think it's  something mistake for  my code 

i'll give images and code-online :
![4919d47381ceab5a3547e1c5a23b23243f448bf4 jpg@253w_150h_1c](https://user-images.githubusercontent.com/30109952/140615105-1d9a95ff-ef3f-464d-af20-50beec30f897.png)

here is code-online demo : 
https://codepen.io/jesson_king/pen/GRvxjoQ

thanks for help !!!

**Comment by adamwathan** [comment, marked as answer]
Hey! iOS Safari doesn't actually focus buttons on click, so you only ever see focus styles for things like inputs and textareas. Just how the OS/browser works — same is actually true in macOS Safari. No way to control it really I'm afraid.

**Comment by XYshop** [comment]
use tabindex="-1"
